core/src/main/java/bisq/core/offer/OpenOfferManager.java

```diff
@@ -221,8 +224,25 @@ public void shutDown(@Nullable Runnable completeHandler) {
         log.info("Remove open offers at shutDown. Number of open offers: {}", size);
         if (offerBookService.isBootstrapped() && size > 0) {
             openOffers.forEach(openOffer -> offerBookService.removeOfferAtShutDown(openOffer.getOffer().getOfferPayload()));
+
+            // wait for offers to be removed, offerBookService::removeOfferAtShutdown may create threads to remove offers...
+            LocalTime timeout = LocalTime.now().plus(Duration.ofSeconds(15));
+            while (!openOffers.getList().isEmpty()) {
+                // check timeout
+                if (timeout.isBefore(LocalTime.now())) {
+                    log.error("Could not remove all offers within timeout (" + openOffers.size() + " offers remaining). Moving on.");
+                    break;
+                }
+                try {
+                    // reduce system load
+                    Thread.sleep(10);
+                } catch (InterruptedException e) {
+                    // ignore
+                }
+            }
+
             if (completeHandler != null)
-                UserThread.runAfter(completeHandler, size * 200 + 500, TimeUnit.MILLISECONDS);
+                UserThread.runAfter(completeHandler, 10, TimeUnit.MILLISECONDS);
```

Comment:
Is there a specific reason why you used 10 milliseconds before triggering the complete handler?

Reply:
In fact after digging into this again, I believe it is best to actually revert this change. We do not have any means of checking whether offers have been taken down already. Syncing that up would mean that we need to alter a whole lot of "stopping" business logic in the code. Although, exactly that is due someday, it is imho not necessary now.